server/controllers/tripsControllers.js

```diff
@@ -11,78 +22,100 @@ class tripscontrolllers {
   if (error)
     return response.response(res, 422,'error', `${error.details[0].message}`, true);
 
-  let trip_origin = await trips.filter(
-    trip_origin =>
-      trip_origin.origin.toUpperCase() === req.body.origin.toUpperCase()
-  );
-  let trip_destination = await trips.filter(
-    trip_destination =>
-      trip_destination.destination.toUpperCase() ===
-      req.body.destination.toUpperCase()
-  );
-  let trip_date = await trips.filter(
-    trip_date => trip_date.trip_date === req.body.trip_date
-  );
-  let trip_time = await trips.filter(
-    trip_time => trip_time.time === req.body.time
-  );
+    let trip_origin = await client.query('SELECT * FROM trips WHERE origin=$1 ',[
+      req.body.origin,
+    ]);
+
+    let trip_destination = await client.query('SELECT * FROM trips WHERE destination=$1 ',[
+      req.body.destination,
```

Comment:
Just like I've pointed above create service files to handle database querying methods. Your controllers are doing a lot of work.

The purpose of a controller is to receive requests pass control to other methods and then return the response to the client